python/cudf/cudf/core/reshape.py

```diff
@@ -372,3 +373,51 @@ def length_check(obj, name):
             df_list.append(col_enc_df)
 
         return concat(df_list, axis=1).drop(labels=columns)
+
+
+def merge_sorted(
+    objs, keys=None, index=False, ascending=True, na_position="last"
+):
+    """Merge a list of sorted DataFrame or Series objects.
+
+    Dataframes/Series in objs list MUST be pre-sorted by columns
+    listed in `keys`, or by the index (if `index=True`).
+
+    Parameters
+    ----------
+    objs : list of DataFrame, Series, or Index
+    keys : list, default None
+        List of Column names to sort by. If None, all columns used
+        (Ignored if `index=True`)
+    index : bool, default False
+        Use index for sorting. `keys` input will be ignored if True
+    ascending : bool, default True
+        Sorting is in ascending order, otherwise it is descending
+    na_position : {‘first’, ‘last’}, default ‘last’
+        'first' nulls at the beginning, 'last' nulls at the end
+
+    Returns
+    -------
+    A new, lexocographically sorted, DataFrame/Series.
+    """
+
+    if not pd.api.types.is_list_like(objs):
+        raise TypeError("objs must be a list-like of Frame-like objects")
+
+    if len(objs) < 1:
+        raise TypeError("objs must be non-empty")
+
+    if len(objs) == 1:
+        return objs[0]
+
+    result = objs[0]._constructor._from_table(
```

Comment:
We've been using `__class__` here as `_from_table` is a class method, and `_constructor` could be confusing in the case of different index classes and whatnot.
```suggestion
    result = objs[0].__class__._from_table(
```